lib/src/annotation.dart

```diff
@@ -150,7 +150,17 @@ dynamic _createFromConstructor(
       // field assigned in the object. Then we can take the field value and
       // set it as the argument value
 
+      // If the annotation class's constructor has no initializers and just
+      // forwards to a `super` class constructor, then there is no need
+      // to add a ctor argument name as a field value.
+      //
+      // In this case `singleWhere` would throw a StateError.
+      if (ctor.constantInitializers.length == 1 && ctor.constantInitializers.first is SuperConstructorInvocation)
+        continue;
```

Comment:
> This happens in a case like the following:
> ```dart
> // Relationship is also an annotation class
> class HasMany extends Relationship {
>  // This `super` call 
>  const HasMany():super();
> }
>```

Why I am confused is because the constructor in the test/example doesn't have any parameters, and this is inside a for loop that is looping over the parameters, so I wouldn't expect it to get executed at all?

> Also, yes, it seems kind of iffy to skip a parameter, but it would always throw an error because the code would cast each member of constructorInitalizers to ConstructorFieldInitializer, which would always throw a type error. I'm hoping there is a better way to do this, but this seems to work.

I don't think we have to support the following, but I want to make sure the error is reasonable:

```dart
class A {
  final String foo;
  const A(this.foo);
}

class B extends A {
 const B(String foo) : super(foo);
}
```

I think today you would just end up bailing out and then not recreating the annotation properly later on, which would be very hard to track down.

This is essentially the same thing as what I wanted to have a test added for :).